core/src/test/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilderTest.java

```diff
@@ -0,0 +1,255 @@
+        GeoBoundingBoxQueryBuilder builder = new GeoBoundingBoxQueryBuilder("mapped_geo");
+        Rectangle box = RandomShapeGenerator.xRandomRectangle(getRandom(), RandomShapeGenerator.xRandomPoint(getRandom()));
+
+        if (frequently()) {
+            // check the top-left/bottom-right combination of setters
+            int path = randomIntBetween(0, 2);
+            switch(path) {
+                case 0: 
+                    builder.topLeft(new GeoPoint(box.getMaxY(), box.getMinX()));
+                    break;
+                case 1:
+                    builder.topLeft(GeohashUtils.encodeLatLon(box.getMaxY(), box.getMinX()));
+                    break;
+                default:
+                    builder.topLeft(box.getMaxY(), box.getMinX());
+            }
+
+            path = randomIntBetween(0, 2);
+            switch(path) {
+            case 0: 
+                builder.bottomRight(new GeoPoint(box.getMinY(), box.getMaxX()));
+                break;
+            case 1:
+                builder.bottomRight(GeohashUtils.encodeLatLon(box.getMinY(), box.getMaxX()));
+            default:
+                builder.bottomRight(box.getMinY(), box.getMaxX());
+            }
+        } else {
+            // check the deprecated top-right/bottom-left combination of setters
+            int path = randomIntBetween(0, 2);
+            switch(path) {
+                case 0: 
+                    builder.topRight(new GeoPoint(box.getMaxY(), box.getMaxX()));
+                    break;
+                case 1:
+                    builder.topRight(GeohashUtils.encodeLatLon(box.getMaxY(), box.getMaxX()));
+                    break;
+                default:
+                    builder.topRight(box.getMaxY(), box.getMaxX());
+            }
+
+            path = randomIntBetween(0, 2);
+            switch(path) {
+            case 0: 
+                builder.bottomLeft(new GeoPoint(box.getMinY(), box.getMinX()));
+                break;
+            case 1:
+                builder.bottomLeft(GeohashUtils.encodeLatLon(box.getMinY(), box.getMinX()));
+            default:
+                builder.bottomLeft(box.getMinY(), box.getMinX());
+            }
+        }
+
+        if (randomBoolean()) {
+            builder.normalize(randomBoolean());
+        }
+
+        // TODO we aren't testing the "memory" scenario - needs a node pulled up for testing
+        builder.type(Type.INDEXED);
+        return builder;
+    }
+
+    private double invalidRandomDouble(double validStart, double validEnd) {
+        Double[] values = new Double[6];
+        values[0] = Double.NaN;
+        values[1] = Double.POSITIVE_INFINITY;
+        values[2] = Double.NEGATIVE_INFINITY;
+        values[3] = randomDoubleBetween(-Double.MAX_VALUE, validStart, true);
+        values[4] = randomDoubleBetween(validEnd, Double.MAX_VALUE, false);
+        values[5] = Double.MAX_VALUE;
+        return randomFrom(values);
```

Comment:
can we add a private static final constant for all the potential values rather than creating a new array all the time?